File: 2572-append-characters-to-string-to-make-subsequence/append-characters-to-string-to-make-subsequence.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        int index = 0;
        for (int i =0;i<s.size();i++){
            if (index < t.size()){
                if (s[i]==t[index]){
                    index++;
                }
            }
        }
    return t.size() - index;
    }
};

File: 2572-append-characters-to-string-to-make-subsequence/append-characters-to-string-to-make-subsequence.dart
class Solution {
  int appendCharacters(String s, String t) {
    int index = 0;
    for (int i =0;i<s.length;i++){
        if (index < t.length){
            if (s[i]==t[index]){
                index++;
            }
        }
    }
    return t.length - index;
  }
}